File: leetcode/js/find-missing-and-repeated-values.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function (grid) {
  const n = grid.length ** 2;
  const count = Array(n + 1).fill(0);

  for (let i = 0; i < grid.length; i++) {
    for (let j = 0; j < grid[0].length; j++) {
      count[grid[i][j]]++;
    }
  }

  const repeat = count.findIndex((n) => n === 2);
  const missing = count.findLastIndex((n) => n === 0);

  return [repeat, missing];
};

File: leetcode/rust/find-missing-and-repeated-values.rs
impl Solution {
    pub fn find_missing_and_repeated_values(grid: Vec<Vec<i32>>) -> Vec<i32> {
        let n = grid.len() * grid.len();
        let mut count = vec![0; n + 1];

        for i in 0..grid.len() {
            for j in 0..grid.len() {
                let num = grid[i][j] as usize;
                count[num] += 1;
            }
        }

        let repeat = count.iter().position(|&n| n == 2).unwrap() as i32;
        let missing = count.iter().rposition(|&n| n == 0).unwrap() as i32;

        vec![repeat, missing]
    }
}
